File: src/algorithms/find-special-integer.js
/**
 * Given a sorted array, finds the only number that occurs more than 25% of the
 * time in the array and returns that number.
 * 
 * @param {number[]} arr - An array of numbers
 * @returns {number} - Returns the number that appears more than 25% of the time
 */
function findSpecialInteger (arr) {
  let count = {};
  let len = arr.length / 4;
  for (const num of arr) {
    count[num] = (count[num] || 0) + 1;
    if (count[num] > len) {
      return num;
    }
  }
}

File: src/algorithms/find-special-integer.py
def find_special_integer(arr):
    """Given an integer array sorted in non-decreasing order, there is exactly one integer in the array that occurs more than 25% of the time, return that integer.

    Args:
        arr (List[int]): A list of integers

    Returns:
        Integer: Returns the element that appears more than 25% of the time
    """
    goal = len(arr) / 4
    elems = {}
    if len(arr) == 1:
        return arr[0]
    for i in range(len(arr)):
        if arr[i] in elems:
            elems[arr[i]] += 1
            if elems[arr[i]] > goal:
                return arr[i]
        else:
            elems[arr[i]] = 1
    return -1